library/src/main/java/com/okta/appauth/android/OktaAppAuth.java

```diff
@@ -595,9 +595,9 @@ private void doInit(final Context context, final OktaAuthListener listener) {
                 Log.e(TAG, "Configuration was invalid: " + mConfiguration.getConfigurationError());
                 listener.onTokenFailure(
                         AuthorizationException.GeneralErrors.INVALID_DISCOVERY_DOCUMENT);
+            } else {
+                mConfiguration.acceptConfiguration();
```

Comment:
I think it is better to use `return` after `listener.onTokenFailure` so we can quit method right after initialization failed. I think we need to terminate method right after failure happened.